# CrunchieMunchies
You work in marketing for a food company myCorps, which is developing a new kind of tasty, wholesome cereal called CrunchieMunchies.

You want to demonstrate to consumers how healthy your cereal is in comparison to other leading brands, so you’ve dug up nutritional data on several different competitors.

Your task is to use NumPy statistical calculations to analyze this data and prove that your CrunchieMunchies is the healthiest choice for consumers.

# Task STEPS
1.First, import numpy.

In [1]:
# your code goes here
import numpy as np

2.Look over the cereal.csv file. This file contains the reported calorie amounts for different cereal brands. Load the data from the file and save it as calorie_stats.

In [3]:
calorie_stats = np.loadtxt(open('cereal.csv','r'),delimiter=',')
calorie_stats

array([ 70., 120.,  70.,  50., 110., 110., 110., 130.,  90.,  90., 120.,
       110., 120., 110., 110., 110., 100., 110., 110., 110., 100., 110.,
       100., 100., 110., 110., 100., 120., 120., 110., 100., 110., 100.,
       110., 120., 120., 110., 110., 110., 140., 110., 100., 110., 100.,
       150., 150., 160., 100., 120., 140.,  90., 130., 120., 100.,  50.,
        50., 100., 100., 120., 100.,  90., 110., 110.,  80.,  90.,  90.,
       110., 110.,  90., 110., 140., 100., 110., 110., 100., 100., 110.])

3.There are 60 calories per serving of CrunchieMunchies. How much higher is the average calorie count of your competition?

Save the answer to the variable average_calories and print the variable to the terminal to see the answer.

In [4]:

# your code goes here

average_calories=calorie_stats.mean()
average_calories

106.88311688311688

4.Does the average calorie count adequately reflect the distribution of the dataset? Let’s sort the data and see.

Sort the data and save the result to the variable calorie_stats_sorted. Print the sorted data to the terminal.

In [5]:
# your code goes here
calorie_stats_sorted = np.sort(calorie_stats)
calorie_stats_sorted


array([ 50.,  50.,  50.,  70.,  70.,  80.,  90.,  90.,  90.,  90.,  90.,
        90.,  90., 100., 100., 100., 100., 100., 100., 100., 100., 100.,
       100., 100., 100., 100., 100., 100., 100., 100., 110., 110., 110.,
       110., 110., 110., 110., 110., 110., 110., 110., 110., 110., 110.,
       110., 110., 110., 110., 110., 110., 110., 110., 110., 110., 110.,
       110., 110., 110., 110., 120., 120., 120., 120., 120., 120., 120.,
       120., 120., 120., 130., 130., 140., 140., 140., 150., 150., 160.])

5.Do you see what I’m seeing? Looks like the majority of the cereals are higher than the mean. Let’s see if the median is a better representative of the dataset.

Calculate the median of the dataset and save your answer to median_calories. Print the median so you can see how it compares to the mean.

In [6]:
# your code goes here
median_calories=np.median(calorie_stats)
median_calories

110.0

6.While the median demonstrates that at least half of our values are over 100 calories, it would be more impressive to show that a significant portion of the competition has a higher calorie count that CrunchieMunchies.

Calculate different percentiles and print them to the terminal until you find the lowest percentile that is greater than 60 calories. Save this value to the variable nth_percentile.

In [7]:
# your code goes here

nth_percentile = np.percentile(calorie_stats,3)
nth_percentile

55.599999999999994

7.While the percentile shows us thatthe majority of the competition has a much higher calorie count, it’s an awkward concept to use in marketing materials.

Instead, let’s calculate the percentage of cereals that have more than 60 calories per serving. Save your answer to the variable more_calories and print it to the terminal

In [8]:
# your code goes here
parts= np.sum(calorie_stats[calorie_stats > 60])
whole = np.sum(calorie_stats)
more_calories = (parts/whole)*100
more_calories

98.17739975698665

8.Wow! That’s a really high percentage. That’s going to be very useful when we promote CrunchieMunchies. But one question is, how much variation exists in the dataset? </em></q>Can we make the generalization that most cereals have around 100 calories or is the spread even greater?

Calculate the amount of variation by finding the standard deviation Save your answer to calorie_std and print to the terminal. How can we incorporate this value into our analysis?



In [10]:
# your code goes here

calorie_std = np.std(calorie_stats)
calorie_std

19.35718533390827

9.Write a short paragraph that sums up your findings and how you think this data could be used to myCorp’s advantage when marketing CrunchieMunchies.

given data is not sufficient but as far as the give data we have to conclude that CrunchieMunchies is one of the healthiest cereal because 98% of cereals have more calorie count (above 60) than CruchieMunchies. We also found that standard deviation is lower than average which means that most of the calories are close to the average calorie count of 100+ and sign of further spread so our product will prove its worth and such analysis will be useful in making marketing strategies.